### PR TITLE
feat: DAIDE parser for any pattern and automatic validation

### PR DESCRIPTION
**Pull request by mjspeck**
The parser can now parse not just messages and arrangements, but any
DAIDE pattern.

Keyword objects are now automatically validated upon instantiation.

Closes: #36, #29

### CODE REVIEW

File: src/daidepp/keywords.py [hunk 1]
```diff
@@ -50,17 +79,17 @@ def __str__(self):
...
         return f"( {self.convoying_unit} ) CVY ( {self.convoyed_unit} ) CTO {self.province}"
 
 
 @dataclass
-class MoveByCVY:
+class MoveByCVY(_DAIDEObject):
     unit: Unit
-    province: PROVINCE
-    province_seas: List[PROVINCE_SEA]
+    province: Province
```
Comment (mjspeck) [inline src/daidepp/keywords.py:91]: Change to location

File: src/daidepp/keywords.py [hunk 2]
```diff
@@ -315,51 +353,54 @@ def __str__(self):
...
+        return f"OCC " + " ".join(unit_str)
 
 
 @dataclass
-class CHO:
+class CHO(_DAIDEObject):
     minimum: int
     maximum: int
-    arrangements: List[ARRANGEMENT]
+    arrangments: List[Arrangement]
```
Comment (mjspeck) [inline src/daidepp/keywords.py:397]: typo, should be 'arrangements'

File: src/daidepp/keywords.py [hunk 3]
```diff
@@ -368,83 +409,83 @@ def __str__(self):
...
 
     def __str__(self):
         return f"INS ( {self.arrangement} )"
 
 
 @dataclass
-class QRY:
-    arrangement: ARRANGEMENT
+class QRY(_DAIDEObject):
+    arrangment: Arrangement
```
Comment (mjspeck) [inline src/daidepp/keywords.py:421]: 'arrangement'

**Comment by byung-o** [comment]
looks good

**Review by byung-o — COMMENTED** [review]
looks good

**Review by byung-o — APPROVED** [review]

**Pull request merged**